packages/gatsby-image/src/index.js

```diff
@@ -224,7 +224,7 @@ class Image extends React.Component {
         this.setState({ isVisible: true }, () =>
           this.setState({
             imgLoaded: imageInCache,
-            imgCached: this.imageRef.current.currentSrc.length > 0,
+            imgCached: !!this.imageRef.current.currentSrc,
```

Comment:
this changes behaviour (`!![]` is true, because empty array is truthy). Adding sanity check seems better here?
```suggestion
            imgCached: this.imageRef.current.currentSrc && this.imageRef.current.currentSrc.length > 0,
```

Reply:
I just noticed this is exactly what you had initially ;)

Reply:
But `currenctSrc` is `DOMString`, not an `Array`. I'm thinking the initial solution was just another way of ensuring it wasn't an empty `string`

Reply:
Either will work, but the `length` check just seems redundant for a string.

Reply:
@thebuilder It had to be a string that was not empty. Sorry for the confusion with how I approached that.

I had unfortunately not tested other cases outside of native IO support with desktop Chrome. I'm pretty sure other paths could cause it to not work correctly, which the new `loading` attribute for native browser lazy-load support may also introduce..

There is a longer more drawn out way to check for a cache, but adds much more delay to be reliable due to Promise / setTimeout to check if the `currentSrc` had been updated since.

Did you confirm that the caching detection was correctly working for when a cache was and was not present? Would it be better to compare to an empty string than the `!!` approach?(which isn't particularly clear at first glance with no comments, had to look up git blame to find this once it caught my eye).

Reply:
The reason i went with `!!` (and we can add this as a comment to clarify), is that it validates that the value not `undefined`, `null` or empty `""`.

I think this is a fine way of checking if the src is cached. You already get an async delay by using the `IntersectionObserver`, no need to increase it.

I'd personally consider it a graceful fallback if the internal caching isn't working 100% in an old browser like IE, as long as it still loads the image.

Reply:
@thebuilder Ah ok, yeah a comment might be helpful to clarify that. Are there cases where it's not an empty string? Or is that just assumed?

> You already get an async delay by using the IntersectionObserver, no need to increase it.

IO is delaying when the main image element is "mounted"/rendered by flipping `isVisible` to true. Prior to that, the`currentSrc` isn't available. The ref that is assigned, isn't attached to anything until that point.

As mentioned, I believe there are some routes/paths in the code where I haven't tested, I don't have IE on hand either. Not a big loss though, just avoids some transition effect / rendering when the image is in the browsers cache(while the other cache logic is for run-time cache).

Thanks for catching this though and fixing it! :)

Reply:
It should always be a (empty) string. But, as the issue is about, IE can have the `currentSrc` as `undefined` instead of an empty string.